ganttproject/src/net/sourceforge/ganttproject/gui/ResourceAssignmentsTableModel.java

```diff
@@ -0,0 +1,179 @@
+package net.sourceforge.ganttproject.gui;
+
+import net.sourceforge.ganttproject.language.GanttLanguage;
+import net.sourceforge.ganttproject.resource.HumanResource;
+import net.sourceforge.ganttproject.task.ResourceAssignment;
+import net.sourceforge.ganttproject.task.ResourceAssignmentMutator;
+import net.sourceforge.ganttproject.task.Task;
+
+import javax.swing.table.AbstractTableModel;
+import java.util.*;
+
+/**
+ * @author Oleg Kushnikov
+ */
+public class ResourceAssignmentsTableModel extends AbstractTableModel {
```

Comment:
There is a few good comments from IDEA about visibility restrictions and simplifying code (yellow ticks in the editor scrollbar)